sdcm/tester.py

```diff
@@ -2584,49 +2587,91 @@ def copy_data_between_tables(self, node, src_keyspace, src_table, dest_keyspace,
+                self.log.debug(
+                    'The estimated total number of rows (%s) exceeds allowed limit (%s). Skipping validation.',
+                    estimated_total_rows_number, LIMIT_TOTAL_ROWS_NUMBER)
+                return
+        self.log.debug('Validate partitions info')
+        partitions_dict_after = self.collect_partitions_info(partitions_attributes=self.partitions_attributes)
+        if partitions_dict_after is not None:
+            self.assertEqual(self.partitions_dict_before,
+                             partitions_dict_after,
+                             msg='Row amount in partitions is not same before and after running of nemesis')
+
+    def collect_partitions_info(self, partitions_attributes: PartitionsValidationAttributes) -> dict | None:
         # Get and save how many rows in each partition.
         # It may be used for validation data in the end of test
-        if not (table_name or primary_key_column):
-            self.log.warning('Can\'t collect partitions data. Missed "table name" or "primary key column" info')
+        if not (partitions_attributes.table_name or partitions_attributes.primary_key_column):
+            TestFrameworkEvent(source=self.__class__.__name__,
+                               message='Can\'t collect partitions data. Missed "table name" or "primary key column" info',
+                               severity=Severity.ERROR).publish()
             return {}
 
         error_message = "Failed to collect partition info. Error details: {}"
         try:
             with self.db_cluster.cql_connection_patient(node=self.db_cluster.nodes[0],
                                                         connect_timeout=600) as session:
                 session.default_consistency_level = ConsistencyLevel.QUORUM
-                pk_list = get_partition_keys(ks_cf=table_name, session=session, pk_name=primary_key_column)
+                pk_list = sorted(get_partition_keys(ks_cf=partitions_attributes.table_name, session=session,
+                                                    pk_name=partitions_attributes.primary_key_column))
         except Exception as exc:  # pylint: disable=broad-except
             TestFrameworkEvent(source=self.__class__.__name__, message=error_message.format(exc),
                                severity=Severity.ERROR).publish()
             return None
 
         # Collect data about partitions' rows amount.
         partitions = {}
+        if partitions_attributes.partition_range_with_data_validation:
+            # Count existing partitions that intersects with partition_range_with_data_validation
+            pk_list = [partition for partition in pk_list if
+                       int(partition) in range(partitions_attributes.partition_start_range,
+                                               partitions_attributes.partition_end_range)]
+        save_into_file_name = partitions_attributes.PARTITIONS_ROWS_BEFORE \
```

Comment:
should this file be added to log collector's list?

Reply:
how this will help us investigating issues?

Reply:
Well, the `assertEqual` cannot print the whole differing dictionaries of 'before' and 'after'.
So in case of an diif issue, we can't say what are the differing numbers.
It currently looks like this:
```
2023-06-07 12:04:19.425: (ThreadFailedEvent Severity.ERROR) period_type=one-time event_id=77a67624-2312-49ee-9acd-0374287fd758: message={111: 5555, 332: 5555, 383: 5555, 314: 5555, 239: 555[1010 chars]5555} != {111: 5555, 412: 10000, 332: 5555, 383: 5555, 314: 55[997 chars]5555}
Diff is 1643 characters long. Set self.maxDiff to None to see it. : Row amount in partitions is not same before and after running of nemesis
```

Reply:
fixed